CMakeLists.txt

```diff
@@ -414,7 +429,8 @@ if(CMAKE_SYSTEM_NAME STREQUAL "AIX")
   # So, create a object library
   add_library(onnx OBJECT ${ONNX_SRCS})
 else()
-  add_library(onnx ${ONNX_SRCS})
+  # onnx target doesn't export symbols and should be built as static lib
+  add_library(onnx STATIC ${ONNX_SRCS})
```

Comment:
can we keep this default meaning let BUILD_SHARED_LIBS control whether we use STATIC or SHARED

what is the problem with building shared lib when build_shared_lib is set?

Reply:
The features (classes/members/functions) from onnx library don't use decorators to control visibility.
It is especially required for Windows - `__declspec`.
So the idea behind it was to block building shared `onnx` if it is not supported.